tests/kvbm_integration/test_nemotron_agg.py

```
@@ -0,0 +1,293 @@
+        f"onboard_h2d={m3.get('kvbm_onboard_blocks_h2d', 0)}"
+    )
+
+    # Phase 4 — determinism check
+    # Note: HMA hybrid model data transfer correctness is a known issue
+    # being tracked separately.  For now, log the difference but don't fail.
+    print("\n=== Phase 4: determinism check ===")
+    if response_1 == response_2:
+        print("✓ Phase 4: responses are deterministic across offload/onboard")
+    else:
+        print(
+            f"⚠ Phase 4: responses differ — HMA onboard data correctness "
```

Ruff (0.15.9)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)